db/migrate/20170704102536_use_deleted_in_containers_tables.rb

```diff
@@ -0,0 +1,55 @@
+class UseDeletedInContainersTables < ActiveRecord::Migration[5.0]
+  class ContainerDefinition < ActiveRecord::Base
+  end
+
+  class ContainerGroup < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class ContainerImage < ActiveRecord::Base
+  end
+
+  class ContainerProject < ActiveRecord::Base
+  end
+
+  class ContainerNode < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Container < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def disconnect_to_soft_delete(model)
+    model.where.not(:deleted_on => nil).update_all("ems_id = old_ems_id")
```

Comment:
> @zeari @zakiva @simon3z
> Is there any scenario where a record could have ems_id set and old_ems_id == nil, but deleted_on set?
> Then this would make a bad situation worse, setting ems_id = old_ems_id = nil.

@cben technically there shouldn't be but I suppose it depends on whether it was possible that a crash left unfinished the updates (if multiple ones and not atomic). Also you're never 100% sure of not having had bugs.

If there's a better logic that we could use to avoid possible mistakes then let's try to improve this (maybe we should use `model.where.not(:deleted_on => nil, :ems_id => nil).update_all("ems_id = old_ems_id")` to be on the safe side?).